### PR TITLE
Fixes natural deletion order flag

### LINKED ISSUE #250: Bug with DiffSyncModelFlags.NATURAL_DELETION_ORDER
### Environment
* DiffSync version:  1.9.0

### Observed Behavior

When you have a parent with no changes that has a child with changes and are using `DiffSyncModelFlags.NATURAL_DELETION_ORDER` on the parent, the child will never be synced.

### Expected Behavior

The child should be synced.

### Steps to Reproduce

Check unit test in #249

### PR DESCRIPTION
**Pull request by Kircheneer**
Fixes #250 - see that for more information.

### CODE REVIEW

File: diffsync/helpers.py [hunk 1]
```diff
@@ -396,7 +398,7 @@ def sync_diff_element(self, element: DiffElement, parent_model: Optional["DiffSy
 
         self.incr_elements_processed()
 
-        if not natural_deletion_order:
+        if not natural_deletion_order or self.action is None:
```
Comment (chadell) [inline diffsync/helpers.py:401]: when `self.action` is None?

Comment (Kircheneer) [reply diffsync/helpers.py:401]: I guess it needs a comment - imagine we have a parent and a child and the parent has no changes, then its action will be `None`. This means that without this change the children will not be processed if the parent doesn't have any changes, even if the children do.

Comment (chadell) [reply diffsync/helpers.py:401]: ok, understood

**Review by chadell — APPROVED** [review]
LGTM

**Comment by Kircheneer** [comment]
Will wait to see if this correctly addresses the issue by installing from this branch and merge only of it does.

File: diffsync/helpers.py (hunk 1 above)
Comment (glennmatthews) [reply diffsync/helpers.py:401]: Just comparing against the case above (`if natural_deletion_order and self.action == DiffSyncActions.DELETE and not skip_children:`), should this be the inverse, i.e `if not natural_deletion_order or self.action != DiffSyncActions.DELETE`?

**Review by chadell — APPROVED** [review]
LGTM

**Pull request merged**